tests/storage/datavolume.go

```diff
@@ -261,47 +259,44 @@ var _ = SIGDescribe("DataVolume Integration", func() {
+			fstatOutput, err := exec.ExecuteCommandOnPod(
+				executorPod,
+				executorPod.Spec.Containers[0].Name,
+				[]string{"stat", "-f", "-c", "%a %s", libstorage.DefaultPvcMountPath},
+			)
+			var freeBlocks, ioBlockSize int64
+			_, err = fmt.Sscanf(fstatOutput, "%d %d", &freeBlocks, &ioBlockSize)
 			Expect(err).ToNot(HaveOccurred())
+			freeSize := freeBlocks * ioBlockSize
 
-			pvcRequestSize := pvc.Spec.Resources.Requests[k8sv1.ResourceStorage]
-			pvCapacitySize := pv.Spec.Capacity[k8sv1.ResourceStorage]
+			vmi := libstorage.RenderVMIWithDataVolume(dataVolume.Name, dataVolume.Namespace)
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 500)
 
-			By("Checking if PV capacity is larger than PVC request")
-			if pvCapacitySize.Cmp(pvcRequestSize) > 0 {
-				Expect(getVirtualSize(vmi, dataVolume)).To(Equal((pvcRequestSize.Value())))
-			} else {
-				volumeStatus := v1.VolumeStatus{}
-				Eventually(func() bool {
-					vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-					for _, volStatus := range vmi.Status.VolumeStatus {
-						if volStatus.Name == "disk0" {
-							volumeStatus = volStatus
-							return true
-						}
+			volumeStatus := v1.VolumeStatus{}
+			Eventually(func() bool {
+				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, volStatus := range vmi.Status.VolumeStatus {
+					if volStatus.Name == "disk0" {
+						volumeStatus = volStatus
+						return true
 					}
-					return false
-				}, 30*time.Second, time.Second).Should(BeTrue(), "Expected VolumeStatus for 'disk0' to be available")
+				}
+				return false
+			}, 30*time.Second, time.Second).Should(BeTrue(), "Expected VolumeStatus for 'disk0' to be available")
 
-				Expect(volumeStatus.PersistentVolumeClaimInfo).ToNot(BeNil())
-				Expect(volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead).ToNot(BeNil())
-				overheadPercentage, err := strconv.ParseFloat(string(*volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead), 64)
-				Expect(err).ToNot(HaveOccurred())
-				overheadSize := float64(pvcRequestSize.Value()) * (1.0 - overheadPercentage)
-				expectedSize := alignImageSizeTo1MiB(int64(overheadSize))
-				Expect(getVirtualSize(vmi, dataVolume)).To(Equal(expectedSize))
-			}
+			Expect(volumeStatus.PersistentVolumeClaimInfo).ToNot(BeNil())
+			Expect(volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead).ToNot(BeNil())
+			overheadPercentage, err := strconv.ParseFloat(string(*volumeStatus.PersistentVolumeClaimInfo.FilesystemOverhead), 64)
+			Expect(err).ToNot(HaveOccurred())
+			overheadSize := int64(math.Ceil((1 - overheadPercentage) * float64(freeSize)))
+			expectedSize := alignImageSizeTo1MiB(int64(overheadSize))
+			Expect(getVirtualSize(vmi, dataVolume)).To(Equal(expectedSize))
```

Comment:
TBH you don't have to go for exact equality, could just assert
```go
Expect(getVirtualSize(vmi, dataVolume)).ToNot(BeNumerically(">", expectedSize))
```

Reply:
Just about separating the concerns of this test so it doesn't have to care about this bit... assuming the overhead stuff is covered elsewhere

Reply:
Not a must at all

Reply:
ack